test/unit/active_model/array_serializer/serialization_test.rb

```diff
@@ -49,7 +49,7 @@ def test_serializer_for_array_returns_appropriate_type
         def object.serializer_class; CustomSerializer; end
 
         assert_equal CustomSerializer, Serializer.serializer_for(object)
-        assert_equal CustomSerializer, Serializer.serializer_for('Custom')
+        assert_nil Serializer.serializer_for('Custom')
```

Comment:
I agree with reverting this.

I either missed or misunderstood it when originally reviewing